### PR TITLE
refactored ServerParse classes

### PR DESCRIPTION
**Pull request by parthjindal50**
Reason:
  Improving code readability by dividing the class and reducing duplicates by reusing methods.
Type:
  Architecture Improvement
Influences：
   - Created new class 'ServerParseValidations' which contains methods to handle parsing validations for 'RwSplitServerParse' and 'ShardingServerParse'. This reduced duplicate code and made the classes easier to understand.
        - Moved many common parsing methods in 'RwSplitServerParse' and 'ShardingServerParse' to the base class 'AbstractServerParse' to remove duplicate code.

### CODE REVIEW

File: src/main/java/com/actiontech/dble/server/parser/RwSplitServerParse.java [hunk 1]
```diff
@@ -72,19 +71,19 @@ public int parse(String stmt) {
...
-                    rt = migrateCheck(stmt, i);
+                    rt = serverParseValidations.migrateCheck(stmt, i);
                     break;
                 case 'O':
                 case 'o':
-                    rt = optimizeCheck(stmt, i);
+                    rt = serverParseValidations.optimizeCheck(stmt, i);
                     break;
                 case 'P':
                 case 'p':
```
Comment (LUAgam) [inline src/main/java/com/actiontech/dble/server/parser/RwSplitServerParse.java:89]: whether prepareCheck is also repeated?

**Review by LUAgam — COMMENTED** [review]
thanks for submitting the code, please sign the CLA, and then can check the code

File: src/main/java/com/actiontech/dble/server/parser/RwSplitServerParse.java (hunk 1 above)
Comment (parthjindal50) [reply src/main/java/com/actiontech/dble/server/parser/RwSplitServerParse.java:89]: Yes, I have extracted that as a common method now.

**Comment by parthjindal50** [comment]
@LUAgam I have signed the CLA, but I don't know why it is still showing as pending.

image

**Review by LUAgam — APPROVED** [review]
ok

**Pull request merged**